Sources/SecureEnclave.swift

```diff
@@ -157,11 +146,20 @@ public final class SecureEnclave {
         if !userPrompt.isEmpty {
             secItemQuery[kSecUseOperationPrompt as String] = userPrompt
         }
-        
-        switch Keychain.string(forKey: key, options: secItemQuery) {
-        case let .success(string):
-            return .success(string)
-            
+
+        return Keychain.string(forKey: key, options: secItemQuery).asSecureEnclaveResult
+    }
+}
+
+// MARK: DataResult Extension
+
+extension SecItem.DataResult where SuccessType: Equatable {
```

Comment:
What do you think about making this an initializer on `SecureEnclave.Result` instead? We can still use an extension to add the `Equatable` restriction, but it's a bit more in line with common Swift design guidelines.